scripts/ci/images/fedora-sanitizers/Dockerfile

```diff
@@ -0,0 +1,50 @@
+FROM fedora:latest
+
+# Install core dev packages
+RUN dnf -y install gcc gcc-c++ gcc-gfortran git make curl file patch \
+        zlib-devel bzip2 bzip2-libs bzip2-devel blosc-devel vim valgrind \
+        zeromq-devel libpng-devel hdf5-devel openmpi-devel hdf5-openmpi-devel \
+        python3-devel python3-numpy python3-mpi4py-openmpi \
+        libasan libtsan libubsan clang llvm-devel libcxx-devel
+
+# Cleanup headers and packages
+RUN dnf clean all
+
+# Install CMake nightly
+WORKDIR /opt/cmake
+RUN curl -L https://cmake.org/files/dev/cmake-3.16.20200111-g8b29a83-Linux-x86_64.tar.gz |tar --strip-components=1 -xzv
+ENV PATH=/opt/cmake/bin:${PATH}
+
+# Installing Kitware Ninja
+WORKDIR /usr/local/bin
+RUN curl -L https://github.com/Kitware/ninja/releases/download/v1.9.0.g99df1.kitware.dyndep-1.jobserver-1/ninja-1.9.0.g99df1.kitware.dyndep-1.jobserver-1_x86_64-linux-gnu.tar.gz | tar --strip-components=1 -xzv
+
+# Building an msan-instrumented libc++
+WORKDIR /opt/llvm
+RUN git clone --branch llvmorg-9.0.0 --depth 1 \
+        https://github.com/llvm/llvm-project.git source
+RUN mkdir -p build/libcxx-msan && \
+    cd build/libcxx-msan && \
+    export \
+        CC=/usr/bin/clang \
+        CXX=/usr/bin/clang++ && \
+    /opt/cmake/bin/cmake \
+        -GNinja \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_INSTALL_PREFIX=/opt/llvm/9.0.0/libcxx-msan \
+        -DCMAKE_INSTALL_RPATH='$ORIGIN' \
+        -DLLVM_ENABLE_PROJECTS="libcxx;libcxxabi" \
+        -DLLVM_USE_SANITIZER=MemoryWithOrigins \
+        -DLIBCXX_ENABLE_ABI_LINKER_SCRIPT=OFF \
+        ../../source/llvm && \
+    ninja install-cxxabi install-cxx
+RUN cd ../.. && rm -rf source build
```

Comment:
This will not remove the build tree cruft, only hide it. Docker images are built in layers, and the cruft will still exist in a lower layer. Instead of `rm`-ing `source` and `build`, it's better to use the `FROM` Docker directive to start a new image, and then `COPY` the needed files into the new image. See [Multi-Stage Builds](https://docs.docker.com/develop/develop-images/multistage-build/) and [my PR](https://github.com/ornladios/ADIOS2/pull/1932/files#diff-63519e5293f296370193560c1789ce55R67) for examples.

Reply:
Yeah, that was leftover from a previous iteration.  I typically push squashed images anyways so it still ends up removed but I've updated this one to have it all in one step.